WalletWasabi.Fluent/ViewModels/Wallets/Send/TransactionPreviewViewModel.cs

```diff
@@ -231,11 +231,18 @@ private async Task OnChangePocketsAsync()
 
 	private async Task OnChangeCoinsAsync()
 	{
+		var selectedCoins = (Transaction?.SpentCoins ?? new List<SmartCoin>()).ToList();
+
 		var selectCoinsDialog =
-			await NavigateDialogAsync(new SelectCoinsDialogViewModel(_walletViewModel));
+			await NavigateDialogAsync(new SelectCoinsDialogViewModel(_walletViewModel, selectedCoins));
 
 		if (selectCoinsDialog.Kind == DialogResultKind.Normal && selectCoinsDialog.Result is { })
 		{
+			if (new HashSet<SmartCoin>(selectedCoins).SetEquals(selectCoinsDialog.Result))
```

Comment:
Use `ToHashSet()` instead:

```suggestion
			if (selectedCoins.ToHashSet().SetEquals(selectCoinsDialog.Result))
```